### PR TITLE
#39 Carousel swipe animation

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-twitch/issues/39#issuecomment-792724138 |
| Decisions | - Add carousel slide animation. <br> - Having responsive size behavior by default instead of hardcoded width and heights as suggested by https://github.com/ripe-tech/ripe-twitch-ui/pull/106#discussion_r589344468 |
| Animated GIF | Peek 2021-03-08 18-21 |

### CODE REVIEW

**Comment by gcandal** [comment]
Some things I believe can be tuned:
- `animation` property (string): whether to use slide or fade, in order to keep backwards compatibility
- `wrap` property that controls whether clicking "next" on the last one will send you to the first. Can default to false
- If wrap=false, makes a "fake slide" animation like:

File: vue/components/ui/molecules/carousel/carousel.vue [hunk 1]
```diff
@@ -254,9 +331,17 @@ export const Carousel = {
     },
     methods: {
         next() {
+            if (this.wrap && this.valueData === this.items.length - 1) {
```
Comment (gcandal) [inline vue/components/ui/molecules/carousel/carousel.vue:334]: !wrap

File: vue/components/ui/molecules/carousel/carousel.vue [hunk 2]
```diff
@@ -254,9 +331,17 @@ export const Carousel = {
...
     methods: {
         next() {
+            if (this.wrap && this.valueData === this.items.length - 1) {
+                this.animation = "slide-right-fake";
+                return;
+            }
             this.valueData = (this.valueData + 1) % this.items.length;
         },
         previous() {
+            if (this.wrap && this.valueData === 0) {
```
Comment (gcandal) [inline vue/components/ui/molecules/carousel/carousel.vue:341]: !wrap

**Review by gcandal — APPROVED** [review]

**Comment by joamag** [comment]
Looking good 👏

**Review by joamag — APPROVED** [review]

**Pull request merged**